core/cmd/shell_local.go

```diff
@@ -996,23 +996,37 @@ func (s *Shell) CleanupChainTables(c *cli.Context) error {
 	if err != nil {
 		return s.errorOut(errors.Wrap(err, "error connecting to the database"))
 	}
-
 	defer db.Close()
 
-	tablesToDeleteFromQuery := `SELECT table_name FROM information_schema.columns WHERE "column_name"=$1;`
+	// some tables with evm_chain_id (mostly job specs) are in public schema
+	tablesToDeleteFromQuery := `SELECT table_name, table_schema FROM information_schema.columns WHERE "column_name"=$1;`
 	// Delete rows from each table based on the chain_id.
 	if strings.EqualFold("EVM", c.String("type")) {
-		var tables []string
-		if err = db.Select(&tables, tablesToDeleteFromQuery, "evm_chain_id"); err != nil {
+		rows, err := db.Query(tablesToDeleteFromQuery, "evm_chain_id")
+		if err != nil {
 			return err
+		} else if rows.Err() != nil {
+			return rows.Err()
 		}
-		for _, tableName := range tables {
+
+		var tableNameAndSchema = make(map[string]string)
```

Comment:
Is it safe to assume table name is unique? Maybe should use a slice of `struct{string,string}` instead, or could you just pre-concatenate them here, and use a `[]string`?

Reply:
makes sense to cover that situation 👍🏼 , fixed